src/commands/deploy.rs

```diff
@@ -125,10 +125,9 @@ impl DeployCommand {
         };
 
         let mut login_connection: LoginConnection = serde_json::from_str(&data)?;
-
-        let expiration_date = DateTime::parse_from_rfc3339(&login_connection.expiration)?;
-        let now: DateTime<Utc> = Utc::now();
-        if now > expiration_date {
+        let expiration = login_connection.expiration.clone();
+        if expiration.is_some() && Utc::now() > DateTime::parse_from_rfc3339(&expiration.unwrap())?
```

Comment:
This should not unwrap.

Reply:
Because `expiration` is an `Option<String>` now, we need `unwrap` to get the underlying `String`? There is a good chance that there is a better way to achieve this. Please let me know.

Also, I have moved this logic to the `has_expired` function, and using a `match` instead of an `unwrap`.

Reply:
@rajatjindal the better way is indeed the `match`, which you've already adopted _grin_

Reply:
I was asking if there was a better way to do this within the if condition

Reply:
Well, you could do the match right there in the condition.  _grin_  But that tends to hide the structure of the program.

Honestly, this kind of lifted fallible operation gets fiddly enough to make putting it in the condition of an `if` a bit hard to read, which is why I often prefer to pull out a function or named variable.  I _think_ `expiration.map(|e| Utc::now() > DateTime.parse(e).unwrap_or(DateTime::MIN_UTC)).unwrap_or_default` might work, though I'd have to think _really really_ hard to be sure about the None and Err cases, and it's far far worse to have in an `if` condition than `is_some/unwrap`!